app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,96 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :set_posts, only: [:index]
+  # GET /posts
+  # GET /posts.json
+  def index
+    #@posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show
+    @posts=Post.find(params[:id])
+    @posts=[@posts]
```

Comment:
To show single post why array?

Reply:
To use the same partial which is used by Index.